blaze-core/src/main/scala/org/http4s/blazecore/IdleTimeoutStage.scala

```diff
@@ -147,7 +147,7 @@ object IdleTimeoutStage {
 
   sealed trait State
   case object Disabled extends State
-  case class Enabled(timeoutTask: Runnable, cancel: Cancelable) extends State
+  final case class Enabled(timeoutTask: Runnable, cancel: Cancelable) extends State
```

Comment:
I like these sorts of changes but aren't we running into bincompat issues?

Reply:
For package-private components we can add mima exclusions. For public APIs I'm adding scalafix exclusions with a note to change them for 1.0.

Reply:
Never played with scalafix, but it would be cool if it could detect the private status of stuff :)

Reply:
That's possible. What would it achieve though? For non-final/sealed/private case classes it's just raising a linter error, not making any changes.

Reply:
Ah, just linting, gotcha. Save the manual exclusion?

Reply:
We still want to disallow plain `case class` for new code though, no?

Reply:
Right, of course. Sorry to be dense 😅

Reply:
Not at all! I might disable that rule for this PR though, to avoid having manual changes mixed up with the automatic ones.

Reply:
What do people think about adding `sealed` to public case classes though? If I understand correctly, that would be source-breaking but not binary-breaking.

Reply:
In general I think we should shy away from source-breaking changes on 0.22/0.23. Also a sealed `case class` as a bincompat workaround for a `final` one seems strange and probably not worth it :)

Reply:
(Btw, I think a more idiomatic way to accomplish the same goal might be a package-private constructor. But still would be source-breaking.)

Reply:
I don't think we should be adding MiMa exceptions for public API that breaks bincompat, but as noted, we permit more if we don't enforce it now.  Is there a way to annotate the legacy ones with something akin to `@nowarn`?

Reply:
> I don't think we should be adding MiMa exceptions for public API that breaks bincompat

Agreed! I’d only want to do that for package-private APIs.

> Is there a way to annotate the legacy ones with something akin to @nowarn?

yes: https://scalacenter.github.io/scalafix/docs/users/suppression.html